CodingTracker.A-Altemara/CodingTracker/Menus/Menu.cs

```diff
@@ -0,0 +1,307 @@
+        return selection;
+    }
+
+    /// <summary>
+    /// Displays a menu and allows the user to view, add, edit, or delete coding session records.
+    /// </summary>
+    public static void SessionsMainMenu()
+    {
+        var continueProgram = true;
+        while (continueProgram)
+        {
+            string selection = DisplaySessionMenu();
+            switch (selection)
+            {
+                case "Exit to Main Menu":
+                    continueProgram = false;
+                    AnsiConsole.WriteLine("Exited Session Menu");
+                    break;
+                case "Add New":
+                    SessionMenu.AddNewEntry(_codingDb);
+                    break;
+                case "Edit Existing":
+                    EditEntry(_codingDb);
+                    break;
+                case "View all Sessions":
+                    ViewRecords(_codingDb);
+                    Console.ReadLine();
+                    break;
+                case "Delete a Coding Session":
+                    DeleteEntry(_codingDb);
+                    break;
+                default:
+                    AnsiConsole.WriteLine("Invalid selection press Enter to try again");
+                    Console.ReadLine();
+                    break;
+            }
+        }
+    }
+
+    /// <summary>
+    /// Displays Goals Menu
+    /// </summary>
+    /// <returns>The selection as a string</returns>
+    private static string DisplayGoalMenu()
+    {
+        // uses Spectre to display console menu
+        Console.Clear();
+        AnsiConsole.Markup("[bold purple]Welcome to your Goals tracker![/]\n");
+        AnsiConsole.Markup("[purple]Please select from the following options[/]\n");
+        var selection = AnsiConsole.Prompt(
+            new SelectionPrompt<string>()
+                .Title("What's your Selection?")
+                .PageSize(5)
+                // .MoreChoicesText("[grey](Move up and down to reveal more options)[/]")
+                .AddChoices([
+                    "Add New Goal", "Edit Existing Goal", "Delete a Goal", "View a goal", "View All Goals",
+                    "Exit to Main Menu"
+                ]));
+        return selection;
+    }
+
+    /// <summary>
+    /// Displays a menu and allows the user to view, add, edit, or delete coding goal records.
+    /// </summary>
+    public static void GoalsMainMenu()
+    {
+        while (true)
+        {
+            string selection = DisplayGoalMenu();
+            switch (selection)
+            {
+                case "Add New Goal":
+                    GoalsMenu.AddNewEntry(_goalsDb);
+                    break;
+                case "Edit Existing Goal":
+                    EditEntry(_goalsDb);
+                    break;
+                case "Delete a Goal":
+                    DeleteEntry(_goalsDb);
+                    break;
+                case "View a goal":
+                    var goal = GoalsMenu.GetGoal(_goalsDb);
```

Comment:
You have a bug here where the user can exit the GetGoal method and then when the program gets to the ShowProgressToGoal method the goal and sessions are null.